Amplify/Core/Support/Amplify+Publisher.swift

```diff
@@ -27,6 +50,52 @@ extension Amplify {
             .handleEvents(receiveCancel: { task.cancel() } )
             .eraseToAnyPublisher()
         }
+        
+        /// Create a Combine Publisher for a given AsyncSequence.
+        ///
+        /// Example Usage
+        /// ```
+        /// let subscription = try await Amplify.API.subscribe(
+        ///     request: .subscription(of: Todo.self, type: .onCreate)
+        /// )
+        ///
+        /// let sink = Amplify.Publisher.create(subscription)
+        ///     .sink { completion in
+        ///         // handle completion
+        ///     } receiveValue: { value in
+        ///         // handle value
+        ///     }
+        /// ```
+        ///
+        /// - Parameter sequence: The AsyncSequence for which to create the Publisher.
+        /// - Returns: The Publisher for the given AsyncSequence.
+        public static func create<Sequence: AsyncSequence>(
+            _ sequence: Sequence
+        ) -> AnyPublisher<Sequence.Element, Error> {
+            let subject = PassthroughSubject<Sequence.Element, Error>()
+            let task = Task {
+                do {
+                    for try await value in sequence {
```

Comment:
this `await` could be waiting a long time before it is able to continue to the next iteration. it should use the `withTaskCancellationHandler` function so that the onCancel handler can be called immediately when the Task is cancelled.

Reply:
Done.  This allows it to immediately stop waiting for values from the sequence in the case of a slow iterator.  I kept an internal check for cancellation inside the loop as well, which prevents it from continuing to process values after cancellation in the case of a fast iterator.